sdk/core/azure-core/CHANGELOG.md

```diff
@@ -9,6 +9,7 @@
 ### Bugs Fixed
 
 - Fixed warning for an unused function in curl.cpp when building the SDK using a version of libcurl older than 7.77.0.
+- Invalidate the token cache within `BearerTokenAuthenticationPolicy` whenever a token request comes back with a 401 response.
```

Comment:
Is this logic being used instead of handling CAE challenges, or do we now have both?

Reply:
Instead of handling  CAE, for now.
We decided not to build support for CAE in C++ without an MSAL.

Reply:
Do we force a retry with the 401 also? If not, won't the current call still return to the client as a `RequestFailedException`?

Reply:
Can you explain a bit more what you mean here?
The behavior around retries and what this policy does after receiving a 401 status code remains **unchanged** here.

The scope of this PR is **only** to invalidate the cached token when 401 is received, so the next attempt will reach cred.GetToken and actually make the attempt to get another token (rather than returning an existing one).

Reply:
I agree that allowing the next call to fetch a new token is better than keeping the potentially revoked token in cache. I just was confirming that nothing about this change would ever allow a CAE challenge request to succeed through cache invalidation alone. So in effect, only customer code that happens to retry on their own would avoid throwing here in a CAE token revocation scenario.

Reply:
That makes sense. Thank for clarifying and double checking the impact on the potential CAE flow. Appreciate it :)